examples/howto/Model_Builder.myst.md

```diff
@@ -0,0 +1,255 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Using ModelBuilder class for deploying PyMC models 
+:::{post} Sep 12, 2022
+:tags: model builder class, model class, linear model
+:category: Beginner
+:author: Shashank Kirtania, Thomas Wiecki
+:::
+
++++
+
+##### Motivation
+
++++
+
+Many users using PyMC face difficulty in deploying or saving their designed PyMC model because deploying/saving/loading a user-created model is a cumbersome task in PyMC. One of the reasons behind this is there is no direct way to save or load a model in PyMC like scikit-learn or TensorFlow. To combat this, We created a `ModelBuilder` class to improve workflow and use direct APIs to build, fit, save, load, predict and more.
+
+The new `ModelBuilder` class allows users to use direct methods to `fit`, `predict`, `save`, `load`. Users can create any model they want, inherit the `ModelBuilder` class, and use predefined methods. <br>
+Let's learn more about using an example <br>
+
++++
+
+Let's see the difference in deployment of models with and without ModelBuilder
+
+initialising the model:
+```
+with pm.Model as model:
+    # Data
+    x = np.linspace(start=0, stop=1, num=100)
+    y = 5 * x + 3
+    y = y + np.random.normal(0, 1, len(x))
+    data = pd.DataFrame({"input": x, "output": y})
+    x = pm.MutableData("x", data["input"].values)
+    y_data = pm.MutableData("y_data", data["output"].values)
+    
+    # Fitting
+    idata = pm.sample(1000,1000)
+    idata.extend(pm.sample_prior_predictive())
+    
+    #posterior predict
+    post_pred = pm.sample_posterior_predictive(idata)
+```
+
++++
+
+To install pymc-experimental we can use pip and get the package to use the ModelBuilder
+
++++
+
+Executing following command in your terminal shall do the job <br>
+`pip install git+https://github.com/pymc-devs/pymc-experimental.git`
+
++++
+
+First, we import libraries. We need to deploy a model
+
+```{code-cell} ipython3
+import arviz as az
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pymc as pm
+```
+
+Now we import the `ModelBuilder` present in the model_builder.py file
+
+```{code-cell} ipython3
+from pymc_experimental.model_builder import ModelBuilder
+```
+
+```{code-cell} ipython3
+%config InlineBackend.figure_format = 'retina'
+RANDOM_SEED = 8927
+np.random.seed(RANDOM_SEED)
```

Comment:
You should create a generator like in the [example](https://docs.pymc.io/en/latest/contributing/jupyter_style.html#code-preamble) and use that to generate the samples from the given seed, using the `numpy.random` namespace directly or `RandomState` classes is discouraged: https://numpy.org/doc/stable/reference/random/legacy.html#functions-in-numpy-random